src/main/webapp/app/complaints/complaint-response.service.ts

```diff
@@ -5,19 +5,65 @@ import { SERVER_API_URL } from 'app/app.constants';
 
 import * as moment from 'moment';
 import { ComplaintResponse } from 'app/entities/complaint-response.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { Exercise } from 'app/entities/exercise.model';
 
 type EntityResponseType = HttpResponse<ComplaintResponse>;
 
 @Injectable({ providedIn: 'root' })
 export class ComplaintResponseService {
     private resourceUrl = SERVER_API_URL + 'api/complaint-responses';
 
-    constructor(private http: HttpClient) {}
+    constructor(private http: HttpClient, private accountService: AccountService) {}
 
-    create(complaintResponse: ComplaintResponse): Observable<EntityResponseType> {
+    /**
+     * Checks if a complaint response is locked for the currently logged in user
+     *
+     * A complaint response is never locked for the creator of the complaint response and for instructors
+     *
+     * @param complaintResponse complaint response to check the lock status for
+     * @param exercise exercise used to find out if currently logged in user is instructor
+     */
+    isComplaintResponseLockedForLoggedInUser(complaintResponse: ComplaintResponse, exercise: Exercise) {
+        if (complaintResponse.isCurrentlyLocked && complaintResponse.submittedTime === undefined) {
+            return !(complaintResponse.reviewer?.login === this.accountService.userIdentity?.login || this.accountService.isAtLeastInstructorForExercise(exercise));
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Checks if the lock on a complaint response is active and if the currently logged in user is the creator of the lock
+     * @param complaintResponse complaint response to check
+     */
+    isComplaintResponseLockedByLoggedInUser(complaintResponse: ComplaintResponse) {
+        return !!(
```

Comment:
you can remove the `!!` as the result is already a boolean

Reply:
Done in 72fed0ffad1de5e7dc5f64debe96b4ba8e016aec